drgn_tools/scsiHosts.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper to print scsi hosts
+"""
+from typing import List
+
+from drgn import container_of
+from drgn import Object
+from drgn import Program
+from drgn.helpers.linux.list import list_for_each_entry
+
+
+def _list_of_scsi_host(prog: Program) -> List[Object]:
+    """
+    Iterates through all scsi hosts and returns a
+    list of scsi devices
+    """
+    try:
+        class_in_private = prog.cache["knode_class_in_device_private"]
+    except KeyError:
+        class_in_private = prog.type("struct device_private").has_member(
+            "knode_class"
+        )
+        prog.cache["knode_class_in_device_private"] = class_in_private
```

Comment:
Don't bother caching this. Looking up a type, and testing whether the type has a member, is not a very expensive task. Plus, this function isn't called repeatedly in a performance-sensitive area, so caching doesn't really make sense.